### ISSUE TITLE
[BUG] Invoke-PnPTenantTemplate provisioning Teams not working?

### ISSUE THREAD

**Issue by BaronSparky** [pnp/powershell#502]
### Reporting an Issue or Missing Feature
ISSUE: Teams templates not provisioning

### Expected behavior 
When using **Invoke-PnPTenantTemplate** with a Teams configuration, need to be able to provision templated Teams

### Actual behavior 
Using the most basic PnP Tenant Template, the Teams component is failing in my environment.

The Office 365 Group is correctly created as is the SharePoint but the command appears to silently quit when "Teamifying" the group.

### Steps to reproduce behavior
Using the attached Tenant Template and using an account that is a Global Administrator:
[SimpleTeam.txt](https://github.com/pnp/powershell/files/6220753/SimpleTeam.txt)

`PS C:\Temp> Connect-PnPOnline -Url https://<myTenant>-admin.sharepoint.com -Credentials $creds`
`PS C:\Temp> Invoke-PnPTenantTemplate -Path "C:\Temp\SimpleTeam.XML" -Parameters @{"TeamTitle" = "A New Team"; "TeamAlias" = "ANewTeam"}`

I get the Office 365 Group created:
![image](https://user-images.githubusercontent.com/11838512/112817483-63d14100-907a-11eb-931f-ac703d3c8eaa.png)

I get the SPO site too...

But nothing appears in Teams :(

I can't even get the **PnPTraceLog** to log anything so can't even see why.

Any ideas???

### What is the version of the Cmdlet module you are running?
PnP.PowerShell 1.4.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by BaronSparky** [comment]
Looks like a timing issue...  
It fails the first time but leaving it for a minute or so, until I can see the Group has been created and then re-running the Invoke-PnPTenantTemplate, it runs through to add the Team.

But, this isn't the expected experience is it?

**Comment by BaronSparky** [comment]
Seems to exhibit the same behaviour as [[BUG] Issue when applying tenant template using AAD app-only access](https://github.com/pnp/powershell/issues/43) but I am using a user identity to connect in this instance.

Also updated the tenant template to specify an Owner for the group as that was noted in the previous article, with a hard coded value.

I still experience the group is not being created until the Group appears first.  So, re-trying, sometimes multiple times, the template does eventually complete successfully but the cmdlet isn't giving me any indication of a failure/incomplete for me to know to re-apply the template.

**Comment by BaronSparky** [comment]
Also believe that this is connected with [[BUG] New-PnPMicrosoft365Group -CreateTeam does not create a team anymore in 1.4.0 #422](https://github.com/pnp/powershell/issues/422)

`New-pnpmicrosoft365group -DisplayName "A New Team9" -Description "Some description" -MailNickname ANewTeam9 -CreateTeam -Force -Owners admin@<myTenant>.onmicrosoft.com `

As running that command hangs for ages, the group does get created but receive the error:
`New-PnPMicrosoft365Group: Empty Payload. JSON content expected.`

**Comment by BaronSparky** [comment]
Tried with 1.5.0.  Problem is still the same. The Group gets created but it isn't "Teamified".

**Comment by FeCo22** [comment]
Same for me. Group gets created but it isn't "teamified". I tried with PnP.PowerShell Version 1.5.0.

I tried via basic authentication (Connect-PnPOnline -Url $Url -Credentials $Credential) and modern authentication (Connect-PnPOnline -Url $Url -ClientId $ClientID -Tenant $Tenant -CertificateBase64Encoded $Base64). 

For basic authentication I get the following error: 
`Invoke-PnPTenantTemplate : PnP.Framework.Sites.SiteCollection.CreateAsync: Could not create team site.`

For modern authentication I get a 403 error response.

I also tried the hardcoded value for the Group Owner Parameter (https://github.com/pnp/powershell/issues/43#issuecomment-755283481) and still failed. 

For me it only works (Workaround) for both authentication methods if I first create the M365 Group:
`New-pnpmicrosoft365group -DisplayName $DisplayName -Description $Description -MailNickname $MailNickName -CreateTeam -Force -IsPrivate -Owners $Owners`

and then Invoke-PnPTenantTemplate
`Invoke-PnPTenantTemplate -Path $Path -Parameters @{"TeamTitle"="$TeamTitle"}`
